### PR TITLE
15 missing implementation of ner module

### LINKED ISSUE #15: Missing implementation of NER module
Located in `src/aiai_eval/named_entity_recognition.py`.

Should follow the same structure as in the `text_classification` module.

Should include unit tests.

### PR DESCRIPTION
**Pull request by AJDERS**
This PR introduces:

* Functionality for NER.
* Tests for NER.
* New exception and corresponding tests.
* Modifications to `test_task.py` such that certain tests are skipped based on the input task fixture.
* Minor changes to sequence classification tests.

Closes #15

### CODE REVIEW

File: tests/test_named_entity_recognition.py [hunk 1]
```diff
@@ -1 +1,132 @@
...
+                "lemmas",
+                "sent_id",
+                "tok_ids",
+                "pos_tags",
+                "morph_tags",
+                "dep_ids",
+                "dep_labels",
+                "ner_tags",
+            ]
+        )
```
Comment (saattrupdan) [inline tests/test_named_entity_recognition.py:107]:
```suggestion
        assert set(tokenised_dataset.features.keys()) == {
            "text",
            "labels",
            "input_ids",
            "token_type_ids",
            "attention_mask",
            "tokens",
            "lemmas",
            "sent_id",
            "tok_ids",
            "pos_tags",
            "morph_tags",
            "dep_ids",
            "dep_labels",
            "ner_tags",
        }
```

File: tests/test_task.py [hunk 2]
```diff
@@ -39,6 +39,17 @@ def task(evaluation_config, task_config):
...
 
+# Allows us to skip tests based on values set in the task fixture.
+@pytest.fixture(autouse=True)
+def skip_if_not_this_task(request, task):
+    if request.node.get_closest_marker("skip_if_not_this_task"):
+        if (
+            request.node.get_closest_marker("skip_if_not_this_task").args[0]
+            != task.task_config.name
+        ):
+            pytest.skip("skipped on this task: {}".format(task.task_config.name))
```
Comment (saattrupdan) [inline tests/test_task.py:50]:
```suggestion
            pytest.skip(f"Skipped for the {task.task_config.name} task.")
```

**Review by saattrupdan — CHANGES_REQUESTED** [review]

File: src/aiai_eval/named_entity_recognition.py [hunk 3]
```diff
@@ -44,7 +48,165 @@ def _preprocess_data_transformer(
             Hugging Face dataset:
                 The preprocessed dataset.
         """
-        return dataset
+        if framework == "spacy":
+            raise InvalidEvaluation(
+                "Evaluation of text predictions for SpaCy models is not yet "
+                "implemented."
+            )
```
[the remark this answers is not in the document]
Comment (AJDERS) [reply src/aiai_eval/named_entity_recognition.py:55]: Ill open a seperate issue :)

**Review by saattrupdan — APPROVED** [review]
LGTM!

**Pull request merged**